pkg/events/derive/derive.go

```diff
@@ -45,6 +49,29 @@ func singleEventDeriveFunc(id events.ID, deriveArgsFunc deriveArgsFunction) even
 	}
 }
 
+// singleTriggeredEventDeriveFunc create an events.DeriveFunction to generate a single derive trace.Event.
+// The event will be created using the original triggering event information, the ID given and the arguments given.
+// The order of the arguments given will match the order in the event definition, so make sure the order match
+// the order of the params in the events.event struct of the event under events.Definitions.
+// If the arguments given is nil, than no event will be derived.
+func singleTriggeredEventDeriveFunc(id events.ID, deriveArgsFunc deriveArgsFunction) events.DeriveFunction {
+	skeleton := makeEventSkeleton(id)
+	return func(event trace.Event) ([]trace.Event, []error) {
+		args, err := deriveArgsFunc(event)
+		if err != nil {
+			return nil, []error{err}
+		}
+		if args == nil {
+			return []trace.Event{}, nil
+		}
+		de, err := newTriggeredEvent(&event, skeleton, args)
+		if err != nil {
+			return []trace.Event{}, []error{err}
+		}
+		return []trace.Event{de}, nil
+	}
```

Comment:
Shouldn't we just add an argument to this function (isTriggered true/false) instead of duplicating the previous function ?

Reply:
I think it would make more sense to do something similar to the `deriveArgsFunction` and pass an `eventBuilder` function. Then we reduce to a single function receiving both (probably the existing `singleEventDeriveFunc`).